source/libs/features.js

```diff
@@ -0,0 +1,176 @@
+// `setTimeout(fn, 0)` ensures this behavior
+setTimeout(() => {
+	onAjaxedPages(async () => {
+		await globalReady; // Match add() behavior to preserve the same order
+		const ajaxContainer = select('#js-repo-pjax-container,#js-pjax-container');
+		if (ajaxContainer) {
+			ajaxContainer.append(<has-rgh/>);
+		}
+	});
+});
+
+// Rule assumes we don't want to leave it pending:
+// eslint-disable-next-line no-async-promise-executor
+const globalReady = new Promise(async resolve => {
+	await safeElementReady('body');
+
+	if (pageDetect.is500()) {
+		return;
+	}
+
+	if (document.body.classList.contains('logged-out')) {
+		console.warn('%cRefined GitHub%c only works when you’re logged in to GitHub.', 'font-weight: bold', '');
+		return;
+	}
+
+	if (select.exists('html.refined-github')) {
+		console.warn('Refined GitHub has been loaded twice. If you didn’t install the developer version, this may be a bug. Please report it to: https://github.com/sindresorhus/refined-github/issues/565');
+		return;
+	}
+
+	document.documentElement.classList.add('refined-github');
+
+	const {customCSS = ''} = await options;
+	if (customCSS.trim().length > 0) {
+		document.head.append(<style>{customCSS}</style>);
+	}
+
+	resolve();
+});
+
+const run = async ({filename, dependencies, init, deinit}) => {
+	if (dependencies.length > 0 && dependencies.every(c => !c())) {
+		await deinit();
+		return;
+	}
+	const {disabledFeatures = '', logging = false} = await options;
+	const log = logging ? console.log : () => {};
+
+	if (disabledFeatures.includes(filename)) {
+		log('↩️', 'Skipping', filename);
+		return;
+	}
+	try {
+		// Features can return `false` if they declare themselves as not enabled
+		if (await init() !== false) {
+			log('✅', filename);
+		}
+	} catch (error) {
+		console.log('❌', filename);
+		console.error(error);
+	}
+};
+
+/**
+ * Register a new feature
+ *
+ * @param {object} definition Information about the feature
+ * @param {string} definition.id  Must match the filename
+ * @param {booleanFunction[]} [definition.dependencies]  Init is called if any of these is true
+ * @param {(callerFunction|Promise)} definition.load     Loading mechanism for the feature
+ * @param {featureFunction}          definition.init     Function that runs the feature
+ * @param {function}                 [definition.deinit] Function that's called when `dependencies` are all false
+ */
+const add = async definition => {
+	await globalReady;
+
+	const {
+		id: filename,
+		dependencies = [], // Default: On all pages
+		load = fn => fn(), // Run it right away
+		init,
+		deinit = () => {}, // Noop
+		...invalidProps
+	} = definition;
+
+	if (Object.keys(invalidProps).length > 0) {
+		throw new Error(`The function "${filename}" was initialized with invalid props: ${Object.keys(invalidProps).join(', ')}`);
+	}
+
+	if (dependencies.some(d => typeof d !== 'function')) {
+		throw new TypeError('Dependencies must be boolean-returning functions');
```

Comment:
The error is meant for developers, to catch it while developing, not at runtime in the store. 

Initially I thought of accepting Boolean as well but it just opens the door to bugs: did you mean to just call it once or `onAjaxedPages`? Because the value may change but if you pass a Boolean it won’t

Reply:
Oh yes, ok that makes sense. 👍